conf/db.c

```diff
@@ -328,15 +333,44 @@ static int confx_allocator_init(struct confx_allocator *alloc,
 		confdb_alloc(alloc, seg, tx, conf_size);
 }
 
+static int confd_btree_kv_put_cb(struct m0_btree_cb  *cb,
+				 struct m0_btree_rec *rec)
+{
+	struct m0_btree_rec     *put_data = cb->c_datum;
+	struct m0_bufvec_cursor  scur;
+	struct m0_bufvec_cursor  dcur;
+	m0_bcount_t              ksize;
+	m0_bcount_t              vsize;
+
+	ksize = m0_vec_count(&put_data->r_key.k_data.ov_vec);
+	vsize = m0_vec_count(&put_data->r_val.ov_vec);
+	if (ksize > m0_vec_count(&rec->r_key.k_data.ov_vec) ||
```

Comment:
As per btree type, it is fixed key variable value size type btree. Is there any case where `ksize` can be greater than `m0_vec_count(&rec->r_key.k_data.ov_vec)` ?

Reply:
While your assumption is correct I am just trying to make sure that the User is not trying to put lengthier key because of a bug in the code. If we allow this then it will corrupt our tree, hence we are returning an error instead of ASSERTing it.